noxfile.py

```diff
@@ -76,3 +121,105 @@ def run_examples(session: Session) -> None:
         for error in errors:
             print(f"- {error}")
         session.error(1)
+
+
+@nox.session(name="performance:test", python=False)
+def performance_tests(session: Session) -> None:
+    """Execute performance tests, assuming a DB already is ready"""
+    command = [
+        "pytest",
+        str(PERFORMANCE_TEST_DIRECTORY),
+        "--benchmark-sort=name",
+        f"--benchmark-json={CURRENT_BENCHMARK}",
+    ]
+
+    session.run(*command)
+
+
+@dataclass
+class Benchmark:
+    filepath: Path
+    benchmark_data: list[dict[str, Any]] = field(init=False)
+
+    def get_benchmark_test(self, fullname: str) -> Optional[ndarray]:
+        match_test = list(
+            filter(lambda x: x["fullname"] == fullname, self.benchmark_data)
+        )
+        if len(match_test) == 1:
+            return np.array(match_test[0]["stats"]["data"])
+        return None
+
+    def set_benchmark_data(self) -> None:
+        file_json = self.filepath.read_text()
+        file_dict = loads(file_json)
+        self.benchmark_data = file_dict["benchmarks"]
+
+    @property
+    def fullname_tests(self) -> set[str]:
+        return {entry["fullname"] for entry in self.benchmark_data}
+
+
+@nox.session(name="performance:check", python=False)
+def performance_check(session: Session) -> None:
+    """Compare previous & current results of performance tests
+
+    In the event of reasonable changes, the benchmark file should be updated
+    by copying the one saved by the CI locally & updating the existing one
+    at test/performance/.benchmarks/0001_benchmark_performance.json.
+
+    Additionally, the value used in this nox session for comparison should be
+    re-evaluated and potentially updated.
+
+    This performance evaluation is used instead of the ones provided by
+    pytest-benchmark as it should more robustly detect when an individual test's
+    performance changes. The pytest-benchmark options as of 0.5.1 focus on overall
+    performance deviations, like +/- 0.1 s or +/- 10% of the mean, which has the
+    underlying assumption that all tests have similar runtime distributions.
+    """
+    # This threshold is used to determine current_benchmark differs from
+    # previous_benchmark. 0.05 is taken for the evaluative criterion of the
+    # pvalue based on common statistical standards when rejecting the null hypothesi
+    # that the runtime data is drawn from the same distribution
+    alpha_threshold = 0.05
```

Comment:
need to fine tune. at least at n=10 runs at 4M, this is not stable.
dunno if we want secondary criteria, but will discuss steps for moving forward